File: ProjectHensel/app/src/main/java/com/example/projecthensel/MainActivity.java
package com.example.projecthensel;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;
import android.app.Activity;
import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.ImageButton;
import android.widget.LinearLayout;
import android.widget.TextView;
import android.widget.Toast;

import java.util.ArrayList;

public class MainActivity extends AppCompatActivity {

    RecyclerView recyclerView;
    ImageButton ImageButton;
    DateAdapter adapter = new DateAdapter();
    Toast toast2;
    String year, month, date, bool, dataString; // Adding 페이지에서 받아오는 값을 저장
    String memo,address, startTime, endTime; // Detail 페이지로 넘길 값을 저장
    private long backKeyPressedTime = 0; // 마지막으로 뒤로가기 버튼을 눌렀던 시간 저장
    int count;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        final int code = 1001; //add Route 수신코드
        ImageButton = findViewById(R.id.editImageButton);
        recyclerView = findViewById(R.id.recyclerView);
        recyclerView.setLayoutManager(new LinearLayoutManager(this));
        final Intent intent = getIntent();
        Intent intent3 = new Intent(getApplicationContext(), DetailRouteActivity.class);

        if(intent.hasExtra("bool"))
       {
           //add Route 페이지에서 데이터 받아오기
           bool = intent.getExtras().getString("bool");
           year = intent.getExtras().getString("year");
           month = intent.getExtras().getString("month");
           date = intent.getExtras().getString("date");
           count = intent.getExtras().getInt("count");
           address = intent.getExtras().getString("address");
           startTime = intent.getExtras().getString("startHour") + ":" + intent.getExtras().getString("startMin") + " AM";
           endTime = intent.getExtras().getString("endHour") + ":" + intent.getExtras().getString("endMin") + " PM";
           memo = intent.getExtras().getString("memo");
           dataString = month + "월 " + date + "일";

           //Main -> Detail 페이지로 데이터 전달
           intent3.putExtra("hasData", "yes");
           intent3.putExtra("yearToDetail", year);
           intent3.putExtra("dataStringToDetail", dataString);
           intent3.putExtra("countToDetail", count);
           intent3.putExtra("addressToDetail", address);
           intent3.putExtra("startTimeToDetail", startTime);
           intent3.putExtra("endTimeToDetail", endTime);
           intent3.putExtra("memoToDetail", memo);

           startActivity(intent3)
                   .setAction(Intent.ACTION_MAIN)
                   .addCategory(Intent.CATEGORY_LAUNCHER)
                   .addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
           recyclerView.setAdapter(adapter);
           adapter.addItem(new Data(dataString, count));
        }

        //이미지 버튼 클릭시 Add 페이지로 이동
            ImageButton.setOnClickListener(new View.OnClickListener(){     //add Route 페이지로 이동
            @Override
            public void onClick(View v) {
                //intent.addFlags(Intent.FLAG_ACTIVITY_NO_ANIMATION);
                Intent intent2 = new Intent(MainActivity.this, RouteAddingActivity.class);
                startActivityForResult(intent2, code);
            }
        });
    }

    @Override
    public void onBackPressed(){

        //2500ms = 2.5sec
        if(System.currentTimeMillis() > backKeyPressedTime + 2500){
            backKeyPressedTime = System.currentTimeMillis();
            toast2 = Toast.makeText(this, "뒤로 가기 버튼을 한 번 더 누르면 종료됩니다.", Toast.LENGTH_SHORT);
            toast2.show();
            return;
        }
        //마지막으로 뒤로 가기 버튼을 눌렀던 시간에 2.5초를 더해 비교 후 2.5초가 지나지 않았으면 종료
        if(System.currentTimeMillis() <= backKeyPressedTime + 2500){
            finish();
        }
    }
}

File: Login/app/src/main/java/com/example/login/MainActivity.kt
package com.example.login

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Toast

import com.android.volley.Response
import com.android.volley.toolbox.Volley
import com.example.login.Volley.LoginRequest

import kotlinx.android.synthetic.main.activity_main.*
import org.json.JSONException
import org.json.JSONObject

class MainActivity : AppCompatActivity() { // 로그인 페이지

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val GoSignUp = Intent(this, SignUpActivity::class.java)

        SignUpButton.setOnClickListener{
            startActivity(GoSignUp)
        }

        LoginButton.setOnClickListener {
            if (IdEditText.text.toString() == "" || PasswordEditText.text.toString() == "") {
                Toast.makeText(this, "아이디 또는 패스워드를 입력하세요", Toast.LENGTH_SHORT).show()
            }
            else {
                //EditText에 있는 값 가져오기
                val userId = IdEditText.text.toString()
                val userPass = PasswordEditText.text.toString()

                val responseListener  = object : Response.Listener<String>{
                        override fun onResponse(response : String){
                            try {
                                println("hongchul" + response)
                                val jsonObject = JSONObject(response)
                                val success = jsonObject.getBoolean("success")
                                if(success){ // 로그인에 성공한 경우
                                    val userId = jsonObject.getString("userID")
                                    val userPass = jsonObject.getString("userPassword")

                                    Toast.makeText(applicationContext, "로그인에 성공했습니다.", Toast.LENGTH_SHORT).show()
                                    val GoSuccess = Intent(applicationContext, SuccessActivity::class.java)
                                    GoSuccess.putExtra("userID", userId)
                                    GoSuccess.putExtra("userPass", userPass)
                                    startActivity(GoSuccess)
                                } else { // 로그인에 실패한 경우
                                    Toast.makeText(applicationContext, "로그인에 실패했습니다.", Toast.LENGTH_SHORT).show()
                                    return
                                }
                            } catch (e : JSONException){
                                e.printStackTrace()
                            }
                        }
                }
                val loginRequest = LoginRequest(userId, userPass, responseListener)
                val queue = Volley.newRequestQueue(this)
                queue.add(loginRequest)
            }
        }
    }
}